### DISCUSSION TITLE
Lotw sync manual sync dead end

### DISCUSSION THREAD

**Discussion by CaptainBalou** [magicbug/Cloudlog#1543] [Q&A] [answered]
Hello all,

I have a question regarding handling Cloudlog together with Lotw. It looks like I see two issues in my setup.

1. New QSOs are not being uploaded to Lotw
2. On manual synching I always end in a dead end in terms of Cloudlog's GUI

Let me explain that a bit.

1. New QSOs are not being uploaded to Lotw
I had two QSOs this morning. I entered them via the "Live QSO" page. Both are _not_ uploaded to Lotw. Do I manually have to mark every QSO to be uploaded to Lotw? It was sent automatically to eQSL though. I don't know if the behaviour is as expected or if something goes wrong. Maybe the second question is depending on this one as well.
The upload to Club Log worked fine with the cron job which was running in the morning.

2. On manual synching I always end in a dead end in terms of Cloudlog's GUI
This is the page I'm starting on.
![grafik](https://user-images.githubusercontent.com/17015872/180402346-5c5c15df-789a-4202-803a-f65efa8139b7.png)

I'm already confused of the status. I would expect to have something like "there is something to transfer". But maybe I'm wrong.

On a manual sync I always end up in this screen.
![grafik](https://user-images.githubusercontent.com/17015872/180401761-e3bc3c6a-da4a-409c-96bd-34cdf3bd6e0b.png)
As you can see it has no Cloudlog menus anymore and all I can do is "go back" in my browser. I don't know what this screen wants to tell me. 😉 Is that a normal behaviour?
I can see the QSO in Lotw directly which looks like this.
![grafik](https://user-images.githubusercontent.com/17015872/180405545-0acda963-66b8-41a9-9f6e-673372971550.png)

In Cloudlog it looks like this.
![grafik](https://user-images.githubusercontent.com/17015872/180405823-185bfb3e-c302-4a4f-8ed3-28fb79907b69.png)
![grafik](https://user-images.githubusercontent.com/17015872/180405904-be93fbe3-32f6-4b05-8ade-3b2f7051f9e4.png)

Does anyone have ideas how I could get all this analysed and ideally solved? 😊 I would appreciate answers or questions of you which could lead me to a solution.

All the best, 73s
Marco

**Comment by magicbug** [comment]
1. New QSOs not being uploaded sounds like the cron task hasn't been set up? missed a step in the install process this should run in the background every x period.
2. Manual sync was never designed to be **used** by humans it was more just output that might be relevant if you need to debug.

**Reply by CaptainBalou** [reply]
Hello Peter,

1. the job is set up and running. If I call it manually the script returns the same output then the "manual sync" button does.
2. Oh, I'm sorry. It looked like the button is meant to be used at least if you want to have a synchronization "now".

**Reply by magicbug** [reply]
If new QSOs arent being uploaded then there's a problem somewhere with the script being ran by the cron task

**Reply by magicbug** [reply]
You can check whats getting uploaded to LoTW by looking at https://lotw.arrl.org/lotwuser/account?accountcmd=activity cloudlog uploads have -cloudlog on the end of the file name

**Reply by CaptainBalou** [reply]
There seems to have been no upload since a long time. Maybe because I just started to log new QSOs yesterday. But there have been no upload. Even after calling the script manually which returned the output mentioned before. Hmm...

<img width="465" alt="grafik" src="https://user-images.githubusercontent.com/17015872/180490030-18b044c8-7ddc-4d74-99a5-d6b7843c43c8.png">

Seems there was just one upload directly after initial import of all QSOs into Cloudlog and then never again.

**Reply by magicbug** [reply]
OK, if you have logged a "new" QSO and the cron for lotw is set up and working then some things borked.

You could turn on logging in /application/config.php and then check if anything is being chucked out when it's being called.

**Reply by CaptainBalou** [reply]
Sounds great. This is what I was searching for. 
Yes, I logged brand new QSOs and the status remains "not sent to LoTW".

Is that this section which would help me here?

```
/*
|--------------------------------------------------------------------------
| Error Logging Threshold
|--------------------------------------------------------------------------
|
| You can enable error logging by setting a threshold over zero. The
| threshold determines what gets logged. Threshold options are:
|
|       0 = Disables logging, Error logging TURNED OFF
|       1 = Error Messages (including PHP errors)
|       2 = Debug Messages
|       3 = Informational Messages
|       4 = All Messages
|
| You can also pass an array with threshold levels to show individual error types
|
|       array(2) = Debug Messages, without Error Messages
|
| For a live site you'll usually only enable Errors (1) to be logged otherwise
| your log files will fill up very fast.
|
*/
$config['log_threshold'] = 0;
```
And do you have a hint where the logs are written to? Apache-Log?

**Reply by CaptainBalou** [reply]
Found some logs in /application/logs/log-* but I can't see errors in there during the upload script runs.

```
INFO - 2022-07-22 22:51:38 --> Config Class Initialized
INFO - 2022-07-22 22:51:38 --> Hooks Class Initialized
INFO - 2022-07-22 22:51:38 --> UTF-8 Support Enabled
INFO - 2022-07-22 22:51:38 --> Utf8 Class Initialized
INFO - 2022-07-22 22:51:38 --> URI Class Initialized
INFO - 2022-07-22 22:51:38 --> Router Class Initialized
INFO - 2022-07-22 22:51:38 --> Output Class Initialized
INFO - 2022-07-22 22:51:38 --> Security Class Initialized
INFO - 2022-07-22 22:51:38 --> Input Class Initialized
INFO - 2022-07-22 22:51:38 --> Language Class Initialized
INFO - 2022-07-22 22:51:38 --> Loader Class Initialized
INFO - 2022-07-22 22:51:38 --> Config file loaded: /var/www/clients/client1/web1/cloudlog/application/config/cloudlog.php
INFO - 2022-07-22 22:51:38 --> Config file loaded: /var/www/clients/client1/web1/cloudlog/application/config/bands.php
INFO - 2022-07-22 22:51:38 --> Config file loaded: /var/www/clients/client1/web1/cloudlog/application/config/lotw.php
INFO - 2022-07-22 22:51:38 --> Helper loaded: url_helper
INFO - 2022-07-22 22:51:38 --> Helper loaded: security_helper
INFO - 2022-07-22 22:51:38 --> Language file loaded: language/english/general_words_lang.php
INFO - 2022-07-22 22:51:38 --> Database Driver Class Initialized
INFO - 2022-07-22 22:51:38 --> Session: Class initialized using 'files' driver.
DEBUG - 2022-07-22 22:51:38 --> cURL Class Initialized
INFO - 2022-07-22 22:51:38 --> Migrations Class Initialized
INFO - 2022-07-22 22:51:38 --> Language file loaded: language/english/migration_lang.php
INFO - 2022-07-22 22:51:38 --> Database Forge Class Initialized
INFO - 2022-07-22 22:51:38 --> Model "Options_model" initialized
INFO - 2022-07-22 22:51:38 --> Controller Class Initialized
INFO - 2022-07-22 22:51:38 --> Helper loaded: form_helper
INFO - 2022-07-22 22:51:38 --> Language file loaded: language/english/lotw_lang.php
INFO - 2022-07-22 22:51:38 --> Model "Stations" initialized
INFO - 2022-07-22 22:51:38 --> Model "LotwCert" initialized
INFO - 2022-07-22 22:51:38 --> Model "User_Model" initialized
INFO - 2022-07-22 22:51:38 --> Model "Logbook_model" initialized
INFO - 2022-07-22 22:51:41 --> Final output sent to browser
INFO - 2022-07-22 22:51:41 --> Total execution time: 2.6830
```

Log level is

```
$config['log_threshold'] = 4;
```

**Reply by CaptainBalou** [reply]
@magicbug Hello Peter. It doesn't log something while running that sync manually through the gui but I saw that I have a log during executing via cron.

```
ERROR - 2022-07-24 19:00:21 --> Severity: Warning --> file_get_contents(https://lotw.arrl.org/lotwuser/lotwreport.adi?login=dk2sf&amp;password=*********&amp;qso_query=1&amp;qso_qsl='yes'&amp;qso_qsldetail='yes'&amp;qso_mydetail='yes'&amp;qso_qslsince=2021-08-25): failed to open stream: HTTP request failed! HTTP/1.1 503 OK
 /var/www/clients/client1/web1/cloudlog/application/controllers/Lotw.php 622
ERROR - 2022-07-24 19:00:21 --> Severity: Warning --> count(): Parameter must be an array or an object that implements Countable /var/www/clients/client1/web1/cloudlog/application/libraries/Adif_parser.php 186
```

Is that something I could work with?

**Reply by CaptainBalou** [reply]
@magicbug Peter, it looks like I've found something by accident. 😉

My station location was wrong. I some day seem to have selected "GERMANY" as my country. But in fact is has to be "FEDERAL REPUBLIC OF GERMANY". I realized that because I tried a manual upload via TQSL.

<img width="920" alt="Bildschirmfoto 2022-07-24 um 23 57 45" src="https://user-images.githubusercontent.com/17015872/180668386-f14ef054-f590-463f-aa50-6df3cfd63ff1.png">


I don't know why that happened. And some years ago it worked but not now. Anyways, I changed that and the new QSOs immediately got synced by a manual sync. The output now also has three lines on the top for my three locations. Awesome.

<img width="1712" alt="grafik" src="https://user-images.githubusercontent.com/17015872/180668468-1fc2c580-e85d-4bd0-b079-fcf03a5a86cf.png">
(screenshot after upload this is why there is 0"No QSOs to upload")

The list beyond that still contains a few QSOs as it did before. I still don't know why they're listed here and what I should do with this information. I would guess these are just the last confirmed QSOs. So the past QSLs I got from others. But if so this should look the same on all users. You sounded like "never seen such an output before" if I got you right.

But anyways. The main issue I had is solved for me. It's syncing again which I want to document here to help the next Germans which possibly did the same mistake. 😉

Thanks and 73s
Marco

**Comment by CaptainBalou** [comment, marked as answer]
Solution:
My station location was wrong. I some day seem to have selected "GERMANY" as my country. But in fact is has to be "FEDERAL REPUBLIC OF GERMANY".
